app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -1916,4 +1922,32 @@ export class Dispatcher {
   ): IDisposable {
     return this.commitStatusStore.subscribe(repository, ref, callback)
   }
+
+  /**
+   * Stashes all changes, including those in untracked files, in the working directory
+   *
+   * @param branch the branch the stash should be associated with
+   */
+  public createStash(repository: Repository, branch: Branch): Promise<void> {
+    return this.appStore._createStash(repository, branch.name)
+  }
+
+  /**
+   * Show the UI for stashed changes
+   */
+  public showStashEntry(repository: Repository) {
+    return this.appStore._showStashEntry(repository)
+  }
+
+  /**
+   * Hide the UI for stashed changes
+   */
+  public hideStashEntry(repository: Repository) {
+    return this.appStore._hideStashEntry(repository)
+  }
+
+  /** Record when the user takes no action on the stash entry */
+  public recordNoActionTakenOnStash(): Promise<void> {
```

Comment:
This function is for you @outofambit. I was thinking you could call it when you click the dismiss button in the diff UI.

Reply:
Or me, depending on merge order...